src/metaxy/metadata_store/_ducklake_support.py

```diff
@@ -0,0 +1,393 @@
+"""Shared DuckLake configuration helpers."""
+
+import os
+from collections.abc import Iterable, Mapping, Sequence
+from dataclasses import dataclass
+from typing import Any, Protocol, runtime_checkable
+
+try:  # pragma: no cover - optional dependency in some environments
```

Comment:
No need to handle import errors. This module should only be imported when the users desides to use ducklake. It should fail if dependencies are missing naturally.